config/opal_setup_cc.m4

```diff
@@ -23,6 +23,119 @@ dnl
 dnl $HEADER$
 dnl
 
+AC_DEFUN([OPAL_CC_HELPER],[
+    AC_MSG_CHECKING([$1])
+
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([$3],[$4])], [$2=yes], [$2=no])
+
+    AC_MSG_RESULT([$$2])
+])
+
+
+AC_DEFUN([OPAL_PROG_CC_C11_HELPER],[
+    OPAL_VAR_SCOPE_PUSH([opal_prog_cc_c11_helper_CFLAGS_save opal_prog_cc_c11_helper__thread_local_available opal_prog_cc_c11_helper_atomic_var_available opal_prog_cc_c11_helper__atomic_available opal_prog_cc_c11_helper_tmp opal_prog_cc_c11_helper__static_assert_available])
+
+    opal_prog_cc_c11_helper_CFLAGS_save=$CFLAGS
+    CFLAGS="$CFLAGS $1"
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 thread local storage], [opal_prog_cc_c11_helper__thread_local_available],
+                   [],[[static _Thread_local int  foo = 1;++foo;]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 atomic variables], [opal_prog_cc_c11_helper_atomic_var_available],
+                   [[#include <stdatomic.h>]], [[static atomic_long foo = 1;++foo;]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 _Atomic keyword], [opal_prog_cc_c11_helper__atomic_available],
+                   [[#include <stdatomic.h>]],[[static _Atomic long foo = 1;++foo;]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 _Generic keyword], [opal_prog_cc_c11_helper__generic_available],
+                   [[#define FOO(x) (_Generic (x, int: 1))]], [[static int x, y; y = FOO(x);]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 _Static_assert], [opal_prog_cc_c11_helper__static_assert_available],
+                   [[#include <stdint.h>]],[[_Static_assert(sizeof(int64_t) == 8, "WTH");]])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper_atomic_var_available" = "xyes"], [opal_prog_cc_c11_helper_tmp=1],
+          [opal_prog_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE_C11_ATOMIC], [$opal_prog_cc_c11_helper_tmp], [Whether C compiler supports C11 atomics])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper__atomic_available" = "xyes"], [opal_prog_cc_c11_helper_tmp=1],
+          [opal_prog_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE__ATOMIC], [$opal_prog_cc_c11_helper_tmp], [Whether C compiler supports __Atomic])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper__thread_local_available" = "xyes"], [opal_prog_cc_c11_helper_tmp=1],
+          [opal_prog_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE__THREAD_LOCAL], [$opal_prog_cc_c11_helper_tmp], [Whether C compiler supports __Thread_local])
+
+    dnl At this time Open MPI only requires thread local and the atomic convenience types
```

Comment:
Open MPI *does not* require these things (yet).

Reply:
Fixed